tips_n_tricks/week3/map-collisions.md

```diff
@@ -0,0 +1,151 @@
+# Map Collisions
+
+## Review
+
+So quick review on collisions, we've covered:
+
+1. distance < half widths  
+![](distance_width.gif). 
+2. standard box-box collision (AABB)
+(gif coming soon)
+3. The pixel perfect [Duck Hunt](https://www.youtube.com/watch?v=-1NyIsZXeqU)-style detection (gif coming soon)
+
+\* here's [a cart (coming soon)](#) that the above gifs came from
+
+## Map Collisions
+
+Now how about colliding with what we have drawn on the map?  
+We could so for each moving object go through each map sprite and and do box-box collision on the two, but that'd be a lot of checks!  
+
+PICO-8 has some convenient functions that let us do collisions with the map a little more easily.  
+These functions are:
+
+* [`mget`](http://pico-8.wikia.com/wiki/Mget): gets the sprite number at the map coordinate given
+* [`fget`](http://pico-8.wikia.com/wiki/Fget): gets the flag(s) set on a sprite number
+
+### [mget(cellx, celly)](http://pico-8.wikia.com/wiki/Mget)
+
+`mget` gives us the sprite number given x and y coordinates.
+Notice though, we need to give it x and y **cell coordinates**. That refers to the x and y coordinates on the bottom left of the map editor when you hover over a sprite on the map.
+
+![](mget.png)
+
+Because each cell coordinate refers to 1 sprite and each sprite is 8x8 pixels, the sprite that the mouse is pointed at in the above picture would be at the **pixel coordinates** of (8, 112) or (**1 x 8**, **14 x 8**).
+
+`mget` returns the sprite number given the cell x and y coordinates, so `mget(1, 14)` would give us `1` in the picture above, because the sprite at coordinate (1,14) is 1.
+
+
+### [fget(n, f)](http://pico-8.wikia.com/wiki/Fget)
+
+Now that we know what sprite is at a particular x and y coordinate, we can check if that sprite has a flag set.   
+
+**Sprite flags** are those little circles below the color and size pickers in the Sprite Editor.
+
+![](fget.png)
+
+The sprite selected in the above picture has **flag 0** turned on.
+If we look at the documentation, we see that `fget(n, f)` takes a sprite number first and then the flag number second. 
+That means to get the flag for the sprite in the above picture, we'd do `fget(1, 0)`, looking at sprite number 1 to see if flag 0 is set.
+
+Flags can be used any way you'd like but in the case of our map collisions, we can say "if a sprite has flag 0 turned on, the player can't go through it. It's a solid object."
+
+That way I can put other sprites on the map and if flag 0 is off, they'll just be part of the background and the player can just walk through it.
```

Comment:
This is the only usage of "I" in the entire document. Dunno if intentional. "We" is used everywhere else.

Reply:
that's perfect actually. I ended up using it verbatim